bin/node-template/node/src/command.rs

```diff
@@ -15,32 +15,35 @@
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 use sp_consensus_aura::sr25519::{AuthorityPair as AuraPair};
-use sc_cli::{VersionInfo, error};
+use sc_cli::{VersionInfo, self};
 use crate::service;
 use crate::chain_spec;
 use crate::cli::Cli;
 
 /// Parse and run command line arguments
-pub fn run(version: VersionInfo) -> error::Result<()> {
+pub fn run(version: VersionInfo) -> sc_cli::Result<()> {
 	let opt = sc_cli::from_args::<Cli>(&version);
 
-	let config = sc_service::Configuration::new(&version);
+	let mut config = sc_service::Configuration::from_version(&version);
 
 	match opt.subcommand {
-		Some(subcommand) => sc_cli::run_subcommand(
-			config,
-			subcommand,
-			chain_spec::load_spec,
-			|config: _| Ok(new_full_start!(config).0),
-			&version,
-		),
-		None => sc_cli::run(
-			config,
-			opt.run,
-			service::new_light,
-			service::new_full,
-			chain_spec::load_spec,
-			&version,
-		)
+		Some(subcommand) => {
+			subcommand.init(&version)?;
+			subcommand.update_config(&mut config, chain_spec::load_spec, &version)?;
+			subcommand.run(
+				config,
+				|config: _| Ok(new_full_start!(config).0),
+			)
+		},
+		None => {
+			opt.run.init(&version)?;
+			opt.run.update_config(&mut config, chain_spec::load_spec, &version)?;
+			opt.run.run(
+				config,
+				service::new_light,
+				service::new_full,
+				&version,
+			)
+		},
```

Comment:
This should be way less error prone than before.

Before: I had put opt.run.shared_params in the subcommand's init and it was a bug.